pkg/operator/k8sutil/service.go

```diff
@@ -92,3 +98,72 @@ func ParseServiceType(serviceString string) v1.ServiceType {
 	}
 	return v1.ServiceType("")
 }
+
+// ExportService exports a k8s service using Multi Cluster Service API client
+func ExportService(ctx context.Context, client *mcsv1Client.Clientset, service *v1.Service) error {
+	serviceExport := &mcsv1a1.ServiceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            service.Name,
+			Namespace:       service.Namespace,
+			OwnerReferences: service.GetOwnerReferences(),
+		},
+	}
+
+	_, err := client.MulticlusterV1alpha1().ServiceExports(service.Namespace).Create(ctx, serviceExport, metav1.CreateOptions{})
+	if err != nil {
+		if kerrors.IsAlreadyExists(err) {
+			logger.Infof("exported service is already created for the service %q", service.Name)
+			return nil
+		}
+		return errors.Wrapf(err, "failed to create exported service %q", service.Name)
+	}
+
+	return nil
+}
+
+// VerifyExportedService verifies if the ServiceExport status conditions to determine if the service was exported correctly
+func VerifyExportedService(ctx context.Context, client *mcsv1Client.Clientset, name, namespace string) error {
+	retryCount := 5
+	serviceExportWaitTime := 10 * time.Second
+
+	for i := 0; i < retryCount; i++ {
+		exportedService, err := client.MulticlusterV1alpha1().ServiceExports(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to get exported service %q", name)
+		}
+
+		var serviceExportError error
+		if len(exportedService.Status.Conditions) == 0 {
+			serviceExportError = fmt.Errorf("failed to determine status of the exported service")
+		} else {
+			for _, condition := range exportedService.Status.Conditions {
```

Comment:
All conditions indicate an error? There isn't a success condition after it's done?

Reply:
MCS API defines 2 conditions.  [Valid and Conflict](https://github.com/kubernetes-sigs/mcs-api/blob/744bb515a7012679e46c90ff117fe273b78b7de8/pkg/apis/v1alpha1/serviceexport.go#L53-L65)
Submariner lighthouse also adds more, that is, [synced](https://github.com/submariner-io/lighthouse/blob/0b0b53abf641114ce99c05bb7ed2615c045f4a7e/pkg/agent/controller/agent.go#L388-L390)

Reply:
The current implementation might not work every time. I'll check again for any better options.

Reply:
Per discussion, perhaps we can query first for the external IP, and then if it doesn't exist yet, query the conditions for details.

Reply:
working on it.

Reply:
done.